### PR TITLE
[SofaCore] Moves MechanicaMatrix out of MultiMatrix.h

### PR DESCRIPTION
**Pull request by damienmarchal**
One class, one file.

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by damienmarchal** [comment]
[ci-build][ci-with-scene-tests][ci-with-regression-tests]

**Comment by damienmarchal** [comment]
[ci-build][with-scene-tests][with-regression-tests]

File: SofaKernel/modules/SofaCore/src/sofa/core/behavior/MechanicalMatrix.h [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+#include <sofa/type/Vec.h>
+#include <iosfwd>
+
+namespace sofa::core::behavior
+{
+
+namespace
+{
+    using sofa::type::Vec;
+}
```
Comment (guparan) [inline SofaKernel/modules/SofaCore/src/sofa/core/behavior/MechanicalMatrix.h:34]: Nice C++ feature I didn't know about :+1:

**Review by guparan — APPROVED** [review]

File: SofaKernel/modules/SofaCore/src/sofa/core/behavior/MechanicalMatrix.h (hunk 1 above)
Comment (damienmarchal) [reply SofaKernel/modules/SofaCore/src/sofa/core/behavior/MechanicalMatrix.h:34]:
```cpp
namespace _mechanicalmatrix_ 
{

}
```

**Pull request merged**